### PR TITLE
Prepare v1.0.0

### PR DESCRIPTION
**Pull request by adamgreig**
I'm opening this PR so we can move discussions forward around v1; I am very happy to have some changes or more feedback before we go ahead with this.

Essentially, this crate is now just `CriticalSection` and `Mutex`, which are widely used and probably won't change without becoming something totally new (e.g. the new mutex-trait crate). There are some known issues (especially CSs don't disable NMI or HardFault) which perhaps could be documented and understood better, but I think are all platform-specific (cortex-m).

In this PR:

* Remove `StaticResource`: it's not been released in any published version so hasn't been proven yet; we could re-add it in a later release or separate crate
* Remove `deny(warnings)` and instead deny them in CI
* Update version to 1.0.0

We'll still want to close off #26. Otherwise I think everything from #22 is covered.

### CODE REVIEW

**Comment by adamgreig** [comment]
I've pushed an update to address a couple of points that came up during reviewing #26 

Outstanding nonconformances:

* We don't implement `Debug` on our types
* We don't set `html_root_url`
* We don't have examples for all items

I'm inclined to go ahead without `Debug` since it doesn't really make sense for these types. I don't know about `html_root_url` and we don't seem to use it anywhere else.

**Comment by nickray** [comment]
Would there be any downside to deriving Debug for CriticalSection and Mutex? It seems quite possible to have a (perhaps temporary) struct containing one, and then you have to implement Debug by hand for e.g. quick debug logging purposes.

**Comment by adamgreig** [comment]
I guess not, I've added that.

**Comment by adamgreig** [comment]
Last call for feedback; I'll ask again in the meeting tomorrow and if no one has any objections let's get this out there.

**Comment by jonas-schievink** [comment]
> I don't know about `html_root_url` and we don't seem to use it anywhere else.

It is required for docs.rs links from dependencies to work correctly. We should use it in every crate (until rustdoc/docs.rs inserts it automatically).

File: CHANGELOG.md [hunk 1]
```diff
@@ -77,7 +75,8 @@ YANKED due to a soundness issue: see v0.2.1 for details
 
 - Initial release
 
-[Unreleased]: https://github.com/japaric/bare-metal/compare/v0.2.5...HEAD
+[Unreleased]: https://github.com/japaric/bare-metal/compare/v1.0.0...HEAD
+[Unreleased]: https://github.com/japaric/bare-metal/compare/v0.2.5...v1.0.0
```
Comment (jonas-schievink) [inline CHANGELOG.md:79]:
```suggestion
[v1.0.0]: https://github.com/japaric/bare-metal/compare/v0.2.5...v1.0.0
```

**Comment by jonas-schievink** [comment]
> Annoying that you have to bump the version every time too.

Yeah. This is the main reason for why I use cargo-release (and the fact that it can bump changelogs automatically too).

**Comment by adamgreig** [comment]
https://crates.io/crates/bare-metal :tada:

**Comment by adamgreig** [comment]
In 5 days it will have been exactly 3 years since v0.1.0!

**Pull request merged**